### PR TITLE
Implement Firebase auth

### PR DESCRIPTION
**Pull request by JoeBouchard**
I spent way too long barking up the wrong tree. After messing with a variety of packages, I ended up settling on the default firebase package. One restriction is I believe it can only be used on client side components, which might put a damper on optimizations.

### CODE REVIEW

File: app/page.js
Comment (zwwuu) [inline app/page.js:1]: maybe merge with https://github.com/Hack-Weekly/tangerine-moose-blog/pull/18 first, so the styling is correct

File: components/signIn.js
Comment (zwwuu) [inline components/signIn.js:1]: will probably need to move this to a context hook like AuthContext. looks good for now.

**Review by zwwuu — CHANGES_REQUESTED** [review]

**Review by thebeatlesphan — COMMENTED** [review]
I pulled your branch and am trying to get it connected to my Firebase but I'm getting errors every corner.
I think I found the new tutorial for NextJS13 that you followed for this.
https://www.freecodecamp.org/news/create-full-stack-app-with-nextjs13-and-firebase/

I was able to get the app connected to Firebase but I can't get the Auth to work.

I keep getting errors with Firebase not being initialized and my firebase_config missing / incorrect. I've tried hardcoding and using dotenv for my config. I have to do some weird stuff to make it through the errors.


March 4 5:43PM

image

I got the auth working on the live db. There's something going on with my .env file.
I'm not sure why referencing it is not working. I npm install dotenv and created a .env file with proper syntax.
I was able to console.log the variables out and they match the SDK but the firebase.js won't take it. 
I had to hardcode my firebaseConfig in to make it work

**Comment by thebeatlesphan** [comment]
I copied your branch and tried to resolve the deployment issue but was unsuccessful.

I think there's something going on with the deployment and firebase configs.
The deployment failed details just takes me to a 404 page.
Everything is looking fine code-wise.

**Comment by zwwuu** [comment]
fixed in https://github.com/Hack-Weekly/tangerine-moose-blog/pull/24

**Pull request closed without merging**